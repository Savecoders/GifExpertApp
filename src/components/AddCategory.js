import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((cats) => [inputValue, ...cats]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit} className='mg-3vw'>
			<input
				type='text'
				placeholder='Search...'
				value={inputValue}
				onChange={handleInputChange}
				className='field'
			/>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
