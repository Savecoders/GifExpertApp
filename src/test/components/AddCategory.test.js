import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('test in <AddCategory/>', () => {
	const setCategories = jest.fn();
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});
	test('show addCategory', () => {
		expect(wrapper).toMatchSnapshot();
	});
	test('change input', () => {
		const input = wrapper.find('input');
		const value = 'hello word';
		input.simulate('change', { target: { value: value } });
		/* expect(wrapper.find('p').text().trim()).toBe(value); */
	});
	test('not resq', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).not.toHaveBeenCalled();
	});
});
