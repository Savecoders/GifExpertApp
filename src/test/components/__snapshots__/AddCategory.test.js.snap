// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AddCategory/> show addCategory 1`] = `
<form
  className="mg-3vw"
  onSubmit={[Function]}
>
  <input
    className="field"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
